### DISCUSSION TITLE
INSERT ON CONFLICT clause support for HQL/Criteria

### DISCUSSION THREAD

**Discussion by beikov** [hibernate/hibernate-orm#5897] [Design Proposals]
Support a `ON CONFLICT` clause similar to PostgreSQL in HQL/Criteria to allow implementing UPSERT semantics.
The idea of the `ON CONFLICT` clause is to allow controlling what should happen when a constraint violation occurs, with the options `DO NOTHING` and `DO UPDATE`.
Most DBs support this natively. For the ones that don't, we can catch the exception, look at the name of the violated constraint, match it and then execute the matching conflict action through a dedicated statement, but emulation could be done as part of a dedicated issue as well.

**Comment by sebersole** [comment]
> but emulation could be done as part of a dedicated issue as well.

This is probably a good path
